### ISSUE TITLE
Proposed DCE change related to dynamic field access

### ISSUE THREAD

**Issue by Simn** [HaxeFoundation/haxe#6996]
Issues like #6994 keep coming up, so I propose the following change to DCE:

If we have any dynamic field access `(d : Dynamic).fieldName`, we keep `fieldName` on all classes that we're keeping anyway.

**Comment by ncannasse** [comment]
I agree, but maybe only in DCE=std ? I would still like a DCE=full to be really full. We can have an additional dce level "all" that applies to all classes but still does this Dynamic bookchecking

**Comment by back2dos** [comment]
I'd like some way of getting full dce and debugging the thing to just get rid of the dynamic access altogether.
